scrapers/base.py

```diff
@@ -43,3 +43,41 @@ def get_html_content(self, url):
             print "Url is not reachable. Error logged on: {}".format(
                 strftime("%Y-%m-%d %H:%M:%S", gmtime()))
             log.info(str(err))
+
+    def local_store(self, data):
+        """Writes content from the website to file.
+        Usage::
+             file name to write data
+        :param_train_data: the filename
+        :rtype: outputs data to a file.
+        """
+        with open(os.path.dirname(__file__) + "/data/news.json",
+                  "r+") as output_file:
+            try:
+                # checks whether there is data in the output_file.
+                existing_data = json.load(output_file)
+            except:
+                existing_data = {'news': []}
+            finally:
+                if type(existing_data) is not dict:
+                    existing_data = {'news': []}
+                    # if data exists in the file, \
+                    # it adds data obtained by the next scraper.
+                existing_data['news'] += data
+        with open(os.path.dirname(__file__) + "/data/news.json",
+                  "w") as output_file:
+            # writes data to the output file.
+            json.dump(existing_data, output_file, indent=2)
+
+    def aws_store(self, site_name, data):
+        """Writes the data to AWS.
+        """
+        try:
+            self.s3.put_object(
+                Bucket='taxclock.codeforkenya.org',
+                ACL='public-read',
+                Key='data/'+site_name+'.json',
+                Body=json.dumps(data))
+        except Exception as err:
+            self.local_store(data)
+            log.info(str(err))
```

Comment:
Storing in AWS and local are significantly different. i.e one stores as each site's son while locally as a single file.

Is there any reason for this?